### DISCUSSION TITLE
Why cron task catalog_product_index_price_reindex_all exists, when catalog_product_price is already fully reindexed 1 hours before from cron task catalogrule_apply_all?

### DISCUSSION THREAD

**Discussion by kanevbg** [OpenMage/magento-lts#2654] [Ideas]
Let's discuss before opening issue.

Does anyone understand why that is the case? I can not find any logic to that.
`\Mage_CatalogRule_Model_Observer::dailyCatalogUpdate` is executed from cron task `catalogrule_apply_all` at 1 am every day, which in turn is hard-coded to run full re-indexing of catalog_product_price index - see `\Mage_CatalogRule_Model_Rule::applyAll`

Than the core has another crontab task `catalog_product_index_price_reindex_all` which runs every day at 2 am. It simply does again the full price re-indexing. What's the point? Do I miss something?

EDIT: If anyone do check and do not find a reason for that, please comment so I know other people did not found a reason. Thanks.

EDIT 2: and there is one more public method intended to trigger full price re-indexing but I did not find any usage of it: `\Mage_CatalogRule_Model_Observer::runCatalogProductPriceReindex`

**Comment by elidrissidev** [comment]
I think the reason is simply because `Mage_CatalogRule_Model_Rule::applyAll` is also called from other contexts like in `Mage_Adminhtml_Promo_CatalogController::applyRulesAction`, so they decided to just put the reindexing in the same method, but I'm not sure.

I wonder if it's okay to just mark the process as `require_index` instead of reindexing immediately, probably worth discussing.

**Comment by kanevbg** [comment]
Yes, that is clear, what is not obvious is why cron task `catalog_product_index_price_reindex_all` should exists at all in that case.

**Reply by elidrissidev** [reply]
Maybe the code for `applyAll` was added before implementing the cron and they were lazy to refactor it so that the reindexing doesn't happen twice in one hour.

**Comment by kanevbg** [comment]
I have production project which is running since this discussion was started and had no complains since then.
If someone wish to take on to implement than in OpenMage the community is likely to benefit significantly for projects with a lot of data.
